src/lib/utils/error-logger.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const logger = {};
+
+const getLogData = (err) => {
+    let logData;
+    if (err.stack) {
+        logData = err.stack;
+    } else if (err.message) {
+        logData = err.message;
+    } else if (err.toString instanceof Function) {
+        logData = err.toString();
+    } else {
+        logData = err;
+    }
+
+    return logData;
+};
+
+module.exports.register = (server, options, next) => {
+
+    //logs error
+    logger.logError = (tags,err) => {
+        server.log(['error'].concat(tags), getLogData(err));
+    };
+
+    //logs and throws error
+    logger.logAndThrowError = (tags,err) => {
```

Comment:
I would say that it is not normally necessary to catch, log and re-throw errors. However, it can be useful in situations like this (where we are actually using errors to carry useful application information from within the call stack up to the route-handler). 

This is similar to a question that we posed to our Ebay Mobile testing guru friend: "should individual utilities within the same code base trust each other completely, or should they test the functionality of their peer utilities as well? In this case, I decided that Authentication failures were too important to rely on the promise chain and route handler to log them, and that they needed to be logged at the point of failure before being passed up the stack.